presto-product-tests/src/main/java/com/facebook/presto/tests/ImpersonationTests.java

```diff
@@ -0,0 +1,95 @@
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+package com.facebook.presto.tests;
+
+import com.facebook.presto.tests.ImmutableTpchTablesRequirements.ImmutableNationTable;
+import com.google.inject.Inject;
+import com.google.inject.name.Named;
+import com.teradata.tempto.ProductTest;
+import com.teradata.tempto.Requires;
+import com.teradata.tempto.hadoop.hdfs.HdfsClient;
+import org.testng.annotations.Test;
+
+import static com.facebook.presto.tests.TestGroups.*;
+import static com.teradata.tempto.query.QueryExecutor.query;
+import static java.lang.String.format;
+import static java.util.Objects.requireNonNull;
+import static org.testng.Assert.assertEquals;
+
+@Requires(ImmutableNationTable.class)
+public class ImpersonationTests
+        extends ProductTest {
+    @Inject(optional = true)
+    private HdfsClient hdfsClient;
+
+    @Inject(optional = true)
+    @Named("databases.presto.jdbc_user")
+    private String prestoJdbcUser;
+
+    @Inject(optional = true)
+    @Named("databases.hive.warehouse_directory_path")
+    private String warehouseDirectoryPath;
+
+    private String getTableLocation(String tablename) {
+        requireNonNull(warehouseDirectoryPath, "databases.hive.warehouse_directory_path is null");
+        requireNonNull(tablename, "Input tablename is null");
+        return warehouseDirectoryPath + tablename;
+    }
+
+    private void checkTableOwner(String tablename, String tablepath, String expectedOwner) {
+        query(format("DROP TABLE IF EXISTS %s", tablename));
+        query(format("CREATE TABLE %s AS SELECT * FROM NATION", tablename));
+        String owner = hdfsClient.getOwner(tablepath);
+        assertEquals(owner, expectedOwner);
+        query(format("DROP TABLE IF EXISTS %s", tablename));
+    }
+
+    // Without impersonation, the table is created with user specified for
+    // -DHADOOP_USER_NAME in jvm.config
+    @Test(groups = {SINGLENODE, PROFILE_SPECIFIC_TESTS})
+    public void testOwnerSingleNode()
+            throws Exception {
+        String tableName = "check_owner_singlenode";
+        String tableLocation = getTableLocation(tableName);
+        checkTableOwner(tableName, tableLocation, "hive");
+    }
+
+    @Test(groups = {SINGLENODE_HDFS_IMPERSONATION, PROFILE_SPECIFIC_TESTS})
+    public void testOwnerSingleNodeHdfsImpersonation()
+            throws Exception {
+        String tableName = "check_owner_singlenode_hdfs_impersonation";
+        requireNonNull(prestoJdbcUser, "databases.presto.jdbc_user is null");
+        String tableLocation = getTableLocation(tableName);
+        checkTableOwner(tableName, tableLocation, prestoJdbcUser);
+    }
+
+    @Test(groups = {SINGLENODE_KERBEROS_HDFS_IMPERSONATION, PROFILE_SPECIFIC_TESTS})
+    public void testOwnerSingleNodeKerberosHdfsImpersonation()
```

Comment:
This test and the `testOwnerSingleNodeHdfsImpersonation` are actually identical. Please remove one of those and rename the other one, say, to `testHdfsImpersonation` or `testHdfsImpersonationEnabled`. Because this is the actual thing we want to test. Checking the owner is only the `way` how we test that.